Add documentation for CA1514 (#37229)
dotnet/docs: docs/fundamentals/code-analysis/quality-rules/ca1514.md

 - CA1514
 - AvoidLengthCalculationWhenSlicingToEndAnalyzer
helpviewer_keywords:
 - CA1514
author: mpidash
dev_langs:
 - CSharp
 - VB
---

# CA1514: Avoid redundant length argument

| Property                            | Value                                          |
|-------------------------------------|------------------------------------------------|
| **Rule ID**                         | CA1514                                         |
| **Title**                           | Avoid redundant length argument                |
| **Category**                        | [Maintainability](maintainability-warnings.md) |
| **Fix is breaking or non-breaking** | Non-breaking                                   |
| **Enabled by default in .NET 7**    | No                                             |

## Cause

A redundant length argument is passed to <xref:System.String.Substring%2A?displayProperty=nameWithType>, <xref:System.Span%601.Slice%2A?displayProperty=nameWithType>, <xref:System.ReadOnlySpan%601.Slice%2A?displayProperty=nameWithType>, or <xref:System.Memory%601.Slice%2A?displayProperty=nameWithType> when slicing to the end of a string or buffer.

## Rule description

> An explicitly calculated length argument can be error-prone and is not needed when slicing to the end of a string or buffer.

Reviewer: ```suggestion
An explicitly calculated length argument can be error-prone and is unnecessary when you're slicing to the end of a string or buffer.
```

> Omitting the length argument results in more readable and maintainable code.

Reviewer: ```suggestion
Code that omits the length argument is more readable and maintainable.
```
